### PR TITLE
FAI-6292: Add paginator over id field for V2

### PR DESCRIPTION
**Pull request by eskrm**

### CODE REVIEW

File: src/graphql/graphql.ts [hunk 1]
```diff
@@ -281,8 +283,14 @@ export function paginatedQuery(query: string): PaginatedQuery {
 }
 
 export function paginatedQueryV2(query: string): PaginatedQuery {
-  return process.env.GRAPHQL_V2_PAGINATOR === 'relay' ?
-    paginatedWithRelayV2(query) : paginateWithOffsetLimitV2(query);
+  switch (process.env.GRAPHQL_V2_PAGINATOR) {
+    case 'relay':
+      return paginatedWithRelayV2(query);
+    case 'where':
```
Comment (tovbinm) [inline src/graphql/graphql.ts:289]: I insist - `whered`

Comment (willmarks) [reply src/graphql/graphql.ts:289]: Actual name is `keyset cursor` if that matters. https://hasura.io/docs/latest/queries/postgres/pagination/#keyset-cursor-based-pagination

File: src/client.ts [hunk 2]
```diff
@@ -324,10 +324,34 @@ export class FarosClient {
...
+          while (hasNextPage) {
+            const data = await self.gqlNoDirectives(graph, query, {
+              limit: pageSize,
+              id,
+              ...Object.fromEntries(args.entries()),
+            });
+            const edges = traverse(data, edgesPath) || [];
+            for (const edge of edges) {
+              yield edge;
+              id = traverse(edge, edgeIdPath);
```
Comment (eskrm) [inline src/client.ts:344]: delete edgeIdPath

File: src/graphql/graphql.ts (hunk 1 above)
Comment (ted-faros) [reply src/graphql/graphql.ts:289]: I'm guessing this was a typo that impacted the prototype?

Comment (eskrm) [reply src/graphql/graphql.ts:289]: Let's name it keyset

File: src/graphql/graphql.ts [hunk 3]
```diff
@@ -427,6 +435,126 @@ function createOperationDefinition(
...
+          // Skip rest of nodes once edges path has been set
+          return false;
+        }
+        edgesPath.push(node.name.value);
+        const whereArg = node.arguments?.find(
+          (n) => n.name.value === 'where'
+        )?.value;
+        return {
+          ...node,
+          arguments: [
```
Comment (tovbinm) [inline src/graphql/graphql.ts:474]: so do you overwrite the existing arguments completely?

File: src/graphql/graphql.ts (hunk 1 above)
Comment (tovbinm) [reply src/graphql/graphql.ts:289]: fine!

Comment (tovbinm) [reply src/graphql/graphql.ts:289]: nerds ))

**Review by ted-faros — APPROVED** [review]

File: test/graphql.test.ts [hunk 4]
```diff
@@ -196,6 +196,17 @@ describe('graphql', () => {
     expect(paginatedQuery.pageInfoPath).toBeEmpty();
   });
 
+  test('paginated where v2 query', async () => {
```
Comment (tovbinm) [inline test/graphql.test.ts:199]: let's also test what comes out of 
```typescript
const query =
      await loadQueryFile('paginated-commits-where-v2.gql');
const paginatedQuery = sut.paginateWithWhereV2(query);
// ... assertions   
```

**Review by willmarks — APPROVED** [review]

File: src/graphql/graphql.ts (hunk 3 above)
Comment (eskrm) [reply src/graphql/graphql.ts:474]: I only keep existing where arg fields

Comment (ted-faros) [reply src/graphql/graphql.ts:474]: if we want to merge where predicates, we can lift [this code](https://github.com/faros-ai/clio/blob/master/server-v2/src/hasura-client.ts#L149) out of clio

File: test/graphql.test.ts (hunk 4 above)
Comment (eskrm) [reply test/graphql.test.ts]: What do you mean?

Comment (tovbinm) [reply test/graphql.test.ts]: We might want to test what happens with existing arguments if they were provided in the query (see - https://github.com/faros-ai/faros-js-client/pull/142#discussion_r1196678700)

**Pull request merged**